sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -81,6 +82,11 @@ public int MaxBatchSize
         /// </summary>
         public bool InvokeProcessorAfterReceiveTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets the initial offset options to apply when processing.
+        /// </summary>
+        public InitialOffsetOptions InitialOffsetOptions { get; set; }
```

Comment:
Should this be readonly?

Reply:
The processor will accept this each time a partition is initialized and is perfectly happy if you use a different value each time.   There may be some benefit to allowing it to be updated.   Off the top of my head, it could be useful to take a "highwater mark" for enqueue time to keep pushing that forward if your workload is ephemeral and events are not interesting X amount of time after they occur. 

Can it be changed in the body of the function or only during the function config?

Reply:
It's more about our `*Options` pattern where the complex options property itself is readonly but properties inside are writeable.

>Can it be changed in the body of the function or only during the function config?

I don't think we correctly react to any option changes outside the initial config as we often copy the values.

Reply:
> Should this be readonly?

I copied this from the currently released API. Let me know if you think it is worth changing.

Reply:
Hm, good point. Considering that there aren't many customers using it I doubt we would be breaking anyone by removing it.

Reply:
Won't it need the public setter for the binding to work?

Reply:
Don't think so.